timed/projects/views.py

```diff
@@ -171,20 +189,44 @@ class TaskAsssigneeViewSet(ReadOnlyModelViewSet):
     filterset_class = filters.TaskAssigneeFilterSet
 
     def get_queryset(self):
-        return models.TaskAssignee.objects.select_related("task", "user")
+        """Don't show task assignees to customers."""
+        user = self.request.user
+
+        queryset = models.TaskAssignee.objects.select_related("task", "user")
+
+        if models.CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
+            return queryset.none()
+
+        return queryset
 
 
 class ProjectAsssigneeViewSet(ReadOnlyModelViewSet):
     serializer_class = serializers.ProjectAssigneeSerializer
     filterset_class = filters.ProjectAssigneeFilterSet
 
     def get_queryset(self):
-        return models.ProjectAssignee.objects.select_related("project", "user")
+        """Don't show project assignees to customers."""
+        user = self.request.user
+
+        queryset = models.ProjectAssignee.objects.select_related("project", "user")
+
+        if models.CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
+            return queryset.none()
+
+        return queryset
 
 
 class CustomerAsssigneeViewSet(ReadOnlyModelViewSet):
     serializer_class = serializers.CustomerAssigneeSerializer
     filterset_class = filters.CustomerAssigneeFilterSet
 
     def get_queryset(self):
-        return models.CustomerAssignee.objects.select_related("customer", "user")
+        """Don't show customer assignees to customers."""
+        user = self.request.user
+
+        queryset = models.CustomerAssignee.objects.select_related("customer", "user")
+
+        if models.CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
```

Comment:
Deny access by default, only allow if explicitly needed. This code only forbids to customers, but everyone else sees all the data